File: app/src/main/java/com/example/islamic/view/TasbeehActivity.java
package com.example.islamic.view;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.TextView; // تم حذف استيراد ImageView
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import com.example.islamic.R;

public class TasbeehActivity extends AppCompatActivity {

    // تعريف المتغيرات للواجهات
    private CardView cardSubhanAllah, cardAlhamdulillah, cardAllahuAkbar, cardAstaghfirullah, cardResetAll;
    private TextView tvSubhanAllahCount, tvAlhamdulillahCount, tvAllahuAkbarCount, tvAstaghfirullahCount;

    // تم تغيير النوع من ImageView إلى TextView
    private TextView tvResetSubhanallah, tvResetAlhamdulillah, tvResetAllahuakbar, tvResetAstaghfirullah;

    // تعريف متغيرات العدادات
    private int subhanAllahCount = 0;
    private int alhamdulillahCount = 0;
    private int allahuAkbarCount = 0;
    private int astaghfirullahCount = 0;

    // تعريف SharedPreferences لحفظ البيانات
    private SharedPreferences sharedPreferences;
    public static final String SHARED_PREFS = "tasbeehSharedPrefs";
    public static final String SUBHANALLAH_KEY = "subhanAllahKey";
    public static final String ALHAMDULILLAH_KEY = "alhamdulillahKey";
    public static final String ALLAHUAKBAR_KEY = "allahuAkbarKey";
    public static final String ASTAGHFIRULLAH_KEY = "astaghfirullahKey";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tasbeeh);

        initializeViews();
        loadData();
        updateViews();
        setupClickListeners();
    }

    private void initializeViews() {
        cardSubhanAllah = findViewById(R.id.card_subhanAllah);
        cardAlhamdulillah = findViewById(R.id.card_alhamdulillah);
        cardAllahuAkbar = findViewById(R.id.card_allahuAkbar);
        cardAstaghfirullah = findViewById(R.id.card_astaghfirullah);

        tvSubhanAllahCount = findViewById(R.id.tv_subhanAllah_count);
        tvAlhamdulillahCount = findViewById(R.id.tv_alhamdulillah_count);
        tvAllahuAkbarCount = findViewById(R.id.tv_allahuAkbar_count);
        tvAstaghfirullahCount = findViewById(R.id.tv_astaghfirullah_count);

        // ربط أزرار التصفير النصية
        tvResetSubhanallah = findViewById(R.id.tv_reset_subhanallah);
        tvResetAlhamdulillah = findViewById(R.id.tv_reset_alhamdulillah);
        tvResetAllahuakbar = findViewById(R.id.tv_reset_allahuakbar);
        tvResetAstaghfirullah = findViewById(R.id.tv_reset_astaghfirullah);

        cardResetAll = findViewById(R.id.card_reset_all);

        sharedPreferences = getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE);
    }

    private void setupClickListeners() {
        // مستمعي النقر على البطاقات لزيادة العداد
        cardSubhanAllah.setOnClickListener(v -> {
            subhanAllahCount++;
            tvSubhanAllahCount.setText(String.valueOf(subhanAllahCount));
        });

        cardAlhamdulillah.setOnClickListener(v -> {
            alhamdulillahCount++;
            tvAlhamdulillahCount.setText(String.valueOf(alhamdulillahCount));
        });

        cardAllahuAkbar.setOnClickListener(v -> {
            allahuAkbarCount++;
            tvAllahuAkbarCount.setText(String.valueOf(allahuAkbarCount));
        });

        cardAstaghfirullah.setOnClickListener(v -> {
            astaghfirullahCount++;
            tvAstaghfirullahCount.setText(String.valueOf(astaghfirullahCount));
        });

        // مستمعي النقر على أزرار التصفير النصية
        tvResetSubhanallah.setOnClickListener(v -> {
            subhanAllahCount = 0;
            tvSubhanAllahCount.setText(String.valueOf(subhanAllahCount));
        });

        tvResetAlhamdulillah.setOnClickListener(v -> {
            alhamdulillahCount = 0;
            tvAlhamdulillahCount.setText(String.valueOf(alhamdulillahCount));
        });

        tvResetAllahuakbar.setOnClickListener(v -> {
            allahuAkbarCount = 0;
            tvAllahuAkbarCount.setText(String.valueOf(allahuAkbarCount));
        });

        tvResetAstaghfirullah.setOnClickListener(v -> {
            astaghfirullahCount = 0;
            tvAstaghfirullahCount.setText(String.valueOf(astaghfirullahCount));
        });

        // مستمع النقر على زر التصفير الكلي
        cardResetAll.setOnClickListener(v -> {
            resetAllCounts();
        });
    }

    private void resetAllCounts() {
        subhanAllahCount = 0;
        alhamdulillahCount = 0;
        allahuAkbarCount = 0;
        astaghfirullahCount = 0;
        updateViews();
    }

    private void updateViews() {
        tvSubhanAllahCount.setText(String.valueOf(subhanAllahCount));
        tvAlhamdulillahCount.setText(String.valueOf(alhamdulillahCount));
        tvAllahuAkbarCount.setText(String.valueOf(allahuAkbarCount));
        tvAstaghfirullahCount.setText(String.valueOf(astaghfirullahCount));
    }

    private void saveData() {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt(SUBHANALLAH_KEY, subhanAllahCount);
        editor.putInt(ALHAMDULILLAH_KEY, alhamdulillahCount);
        editor.putInt(ALLAHUAKBAR_KEY, allahuAkbarCount);
        editor.putInt(ASTAGHFIRULLAH_KEY, astaghfirullahCount);
        editor.apply();
    }

    private void loadData() {
        subhanAllahCount = sharedPreferences.getInt(SUBHANALLAH_KEY, 0);
        alhamdulillahCount = sharedPreferences.getInt(ALHAMDULILLAH_KEY, 0);
        allahuAkbarCount = sharedPreferences.getInt(ALLAHUAKBAR_KEY, 0);
        astaghfirullahCount = sharedPreferences.getInt(ASTAGHFIRULLAH_KEY, 0);
    }

    @Override
    protected void onPause() {
        super.onPause();
        saveData();
    }
}

File: app/src/main/java/com/eslamdev/islamic/presentation/ui/activity/TasbeehActivity.kt
package com.eslamdev.islamic.presentation.ui.activity

import android.content.Context
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.eslamdev.islamic.R
import com.eslamdev.islamic.databinding.ActivityTasbeehBinding

class TasbeehActivity : AppCompatActivity() {

    private lateinit var binding: ActivityTasbeehBinding

    private val PREFS_NAME = "tasbeehSharedPrefs"
    private val KEY_SUBHAN = "subhanAllahKey"
    private val KEY_HAMD = "alhamdulillahKey"
    private val KEY_AKBAR = "allahuAkbarKey"
    private val KEY_ASTAGHFIR = "astaghfirullahKey"

    private var subhanCount = 0
    private var hamdCount = 0
    private var akbarCount = 0
    private var astaghfirCount = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityTasbeehBinding.inflate(layoutInflater)
        setContentView(binding.root)

       binding.btnBack.setOnClickListener {
            finish()
        }

        loadCounts()
        updateUI()

        binding.cardSubhanView.setOnClickListener {
            subhanCount++
            saveCount(KEY_SUBHAN, subhanCount)
            updateUI()
        }

        binding.cardAlhamdView.setOnClickListener {
            hamdCount++
            saveCount(KEY_HAMD, hamdCount)
            updateUI()
        }

        binding.cardAkbarView.setOnClickListener {
            akbarCount++
            saveCount(KEY_AKBAR, akbarCount)
            updateUI()
        }

        binding.cardAstaghfarView.setOnClickListener {
            astaghfirCount++
            saveCount(KEY_ASTAGHFIR, astaghfirCount)
            updateUI()
        }

        binding.tvResetSubhanallah.setOnClickListener {
            subhanCount = 0
            saveCount(KEY_SUBHAN, 0)
            updateUI()
        }

        binding.tvResetAlhamdulillah.setOnClickListener {
            hamdCount = 0
            saveCount(KEY_HAMD, 0)
            updateUI()
        }

        binding.tvResetAllahuakbar.setOnClickListener {
            akbarCount = 0
            saveCount(KEY_AKBAR, 0)
            updateUI()
        }

        binding.tvResetAstaghfirullah.setOnClickListener {
            astaghfirCount = 0
            saveCount(KEY_ASTAGHFIR, 0)
            updateUI()
        }

        binding.cardResetAll.setOnClickListener {
            subhanCount = 0
            hamdCount = 0
            akbarCount = 0
            astaghfirCount = 0

            val prefs = getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            prefs.edit().clear().apply()

            updateUI()
        }
    }

    private fun updateUI() {
        binding.tvSubhanAllahCount.text = subhanCount.toString()
        binding.tvAlhamdulillahCount.text = hamdCount.toString()
        binding.tvAllahuAkbarCount.text = akbarCount.toString()
        binding.tvAstaghfirullahCount.text = astaghfirCount.toString()
    }

    private fun loadCounts() {
        val prefs = getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        subhanCount = prefs.getInt(KEY_SUBHAN, 0)
        hamdCount = prefs.getInt(KEY_HAMD, 0)
        akbarCount = prefs.getInt(KEY_AKBAR, 0)
        astaghfirCount = prefs.getInt(KEY_ASTAGHFIR, 0)
    }

    private fun saveCount(key: String, value: Int) {
        val prefs = getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        prefs.edit().putInt(key, value).apply()
    }
}